buildkite/resource_cluster_test.go

```diff
@@ -9,105 +9,130 @@ import (
+				ProtoV6ProviderFactories: protoV6ProviderFactories(),
+				CheckDestroy:             testAccCheckClusterDestroy,
+				Steps: []resource.TestStep{
+					{
+						Config: config,
+						Check:  check,
+					},
+				},
+			})
+		}
+		t.Run("with basic settings", func(t *testing.T) {
+			testCase(t, basic(randName))
+		})
+
+		t.Run("with complex settings", func(t *testing.T) {
+			testCase(t, complex(randName))
+		})
 	})
-}
 
-func TestAccCluster_Update(t *testing.T) {
-	resName := acctest.RandString(12)
-	newResName := acctest.RandString(15)
-	t.Parallel()
-	var c = new(clusterResourceModel)
-
-	resource.Test(t, resource.TestCase{
-		PreCheck:                 func() { testAccPreCheck(t) },
-		ProtoV6ProviderFactories: protoV6ProviderFactories(),
-		CheckDestroy:             testAccCheckClusterDestroy,
-		Steps: []resource.TestStep{
-			{
-				Config: testAccClusterBasic(resName),
-				Check: resource.ComposeAggregateTestCheckFunc(
-					testAccCheckClusterExists("buildkite_cluster.foo", c),
-					testAccCheckClusterRemoteValues(c, fmt.Sprintf("%s_test_cluster", resName)),
-					resource.TestCheckResourceAttr("buildkite_cluster.foo", "name", fmt.Sprintf("%s_test_cluster", resName)),
-					resource.TestCheckResourceAttrSet("buildkite_cluster.foo", "id"),
-					resource.TestCheckResourceAttrSet("buildkite_cluster.foo", "uuid"),
-				),
-			},
-			{
-				Config: testAccClusterBasic(newResName),
-				Check: resource.ComposeAggregateTestCheckFunc(
-					testAccCheckClusterExists("buildkite_cluster.foo", c),
-					testAccCheckClusterRemoteValues(c, fmt.Sprintf("%s_test_cluster", newResName)),
-					resource.TestCheckResourceAttr("buildkite_cluster.foo", "name", fmt.Sprintf("%s_test_cluster", newResName)),
-				),
-			},
-		},
+	t.Run("Updates a Cluster", func(t *testing.T) {
+		randName := acctest.RandString(5)
+		randNameUpdated := acctest.RandString(5)
+		testCase := func(t *testing.T, originalConfig, newConfig string) {
+			resource.ParallelTest(t, resource.TestCase{
+				PreCheck:                 func() { testAccPreCheck(t) },
+				ProtoV6ProviderFactories: protoV6ProviderFactories(),
+				CheckDestroy:             testAccCheckClusterDestroy,
+				Steps: []resource.TestStep{
+					{
+						Config: originalConfig,
+						Check: resource.ComposeAggregateTestCheckFunc(
+							testAccCheckClusterExists("buildkite_cluster.foo", &c),
+							testAccCheckClusterRemoteValues(&c, fmt.Sprintf("%s_test_cluster", randName)),
+							resource.TestCheckResourceAttr("buildkite_cluster.foo", "name", fmt.Sprintf("%s_test_cluster", randName)),
+							resource.TestCheckResourceAttrSet("buildkite_cluster.foo", "id"),
+							resource.TestCheckResourceAttrSet("buildkite_cluster.foo", "uuid"),
+						),
+					},
+					{
+						Config: newConfig,
+						Check: resource.ComposeAggregateTestCheckFunc(
+							testAccCheckClusterExists("buildkite_cluster.foo", &c),
+							testAccCheckClusterRemoteValues(&c, fmt.Sprintf("%s_test_cluster", randNameUpdated)),
```

Comment:
This only checks the name but I think it should also check the other properties to make sure its all working correctly

Reply:
With the `basic` config, there is only the `name` set. It also checks that the ID and UUID are set.

Reply:
Added additional to ensure unchanged remain `null`